src/styles/auto-prefix.js

```diff
@@ -3,61 +3,97 @@ import warning from 'warning';
 
 const prefixers = {};
 
+let hasWarnedAboutNavigator = false;
+
+function getPrefixer(userAgent) {
+  if (typeof navigator !== 'undefined') {
+    userAgent = navigator.userAgent;
+  }
+
+  if (userAgent === null && !hasWarnedAboutNavigator) {
+    warning(false, `Material-UI: userAgent should be supplied in the muiTheme context
+      for server-side rendering.`);
+
+    hasWarnedAboutNavigator = true;
+
+    return null;
+  }
+
+  // Get prefixing instance for this user agent
+  let prefixer = prefixers[userAgent];
```

Comment:
I think there is no reason to cache the prefixer. Is building one really expensive?